lib/ocpp/v201/charge_point_callbacks.cpp

```diff
@@ -77,6 +77,22 @@ bool Callbacks::all_callbacks_valid(std::shared_ptr<DeviceModel> device_model) c
         }
     }
 
+    if (valid) {
+        if (device_model->get_optional_value<bool>(ControllerComponentVariables::ReservationCtrlrAvailable)
+                .value_or(false)) {
+            if (!this->reserve_now_callback.has_value() or this->reserve_now_callback == nullptr) {
+                EVLOG_error << "Reservation is set to 'Available' and 'Enabled' in device model, but "
+                               "reserve_now_callback is not implemented.";
+            }
+
+            if (!this->cancel_reservation_callback.has_value() or this->cancel_reservation_callback == nullptr) {
+                EVLOG_error
+                    << "Reservation is set to 'Available' and 'Enabled' in device model, but cancel_reservation "
+                       "callback is not implemented";
+            }
+        }
+    }
+
```

Comment:
This looks like it could be moved inside the upper `if (valid)` and it does not set the `valid` variable to false in case of the error logs

Reply:
Ah good catch

I put it in this `if (valid)` so it will not continue if the previous were not valid. But maybe it is a good idea to have a message of all not valid callbacks instead of fixing one and getting an error about the next. So changed it accordingly as you asked indeed :).